common/src/main/java/io/druid/collections/ReferenceCountingResourceHolder.java

```diff
@@ -103,6 +110,10 @@ public void close()
     };
   }
 
+  /**
+   * Closes this holder and decrements the reference count by 1. This method should be called after all
```

Comment:
That's true. But, it's quite confused to me that the resource can be valid even after the holder is closed. Also, I think this is quite error-prone. For example, if someone forget to close `Releaser` or `ReferenceCountingResourceHolder`, the `closer` in holder will never be called. Probably adding some restrictions to this helps to reduce such mistakes. For example, if we enforce to close the holder after all `Releaser` is closed, we can add a sanity check to `close()` method. What do you think? This might not be an issue of this PR. If you agree, I'll raise another PR for this.